airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactory.kt

```diff
@@ -323,9 +333,11 @@ class ObjectStoragePathFactory(
 
     private fun getFormattedPath(
         stream: DestinationStream,
-        variables: List<PathVariable> = PATH_VARIABLES
+        variables: List<PathVariable> = PATH_VARIABLES,
+        isStaging: Boolean
     ): String {
-        val pattern = pathPatternResolved
+        val selectedPrefix = if (isStaging) stagingPrefix else finalPrefix
+        val pattern = resolveRetainingTerminalSlash(selectedPrefix, pathPatternResolved)
```

Comment:
Wouldn't it be simpler to pass this pattern in instead of passing down a boolean to replicate the branching?